homeassistant/components/config/automation.py

```diff
@@ -107,3 +122,129 @@ async def websocket_automation_trace(hass, connection, msg):
         }
 
     connection.send_result(msg["id"], automation_traces)
+
+
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "automation/debug/breakpoint/set",
+        vol.Required("automation_id"): str,
+        vol.Required("node"): str,
+        vol.Optional("run_id"): str,
+    }
+)
+@websocket_api.async_response
+async def websocket_automation_breakpoint_set(hass, connection, msg):
+    """Set breakpoint."""
+    automation_id = msg.get("automation_id")
+    node = msg.get("node")
+    run_id = msg.get("run_id")
+
+    result = breakpoint_set(hass, automation_id, run_id, node)
+
+    connection.send_result(msg["id"], result)
+
+
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "automation/debug/breakpoint/clear",
+        vol.Required("automation_id"): str,
+        vol.Required("node"): str,
+        vol.Optional("run_id"): str,
+    }
+)
+@websocket_api.async_response
+async def websocket_automation_breakpoint_clear(hass, connection, msg):
+    """Set breakpoint."""
+    automation_id = msg.get("automation_id")
+    node = msg.get("node")
+    run_id = msg.get("run_id")
+
+    result = breakpoint_set(hass, automation_id, run_id, node)
+
+    connection.send_result(msg["id"], result)
+
+
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "automation/debug/breakpoint/list",
+    }
+)
+@websocket_api.async_response
+async def websocket_automation_breakpoint_list(hass, connection, msg):
```

Comment:
No need to be async.

```suggestion
def websocket_automation_breakpoint_list(hass, connection, msg):
```

Reply:
Fixed in bc24777